File: src/main/java/ru/dargen/evoplus/event/chat/ChatReceiveEvent.java
package ru.dargen.evoplus.event.chat;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import net.minecraft.text.Text;
import ru.dargen.evoplus.event.CancellableEvent;

@Getter @Setter
@AllArgsConstructor
public class ChatReceiveEvent extends CancellableEvent {

    protected Text text;

    public void setText(String text) {
        this.text = Text.of(text);
    }

}

File: src/main/kotlin/ru/dargen/evoplus/event/chat/ChatReceiveEvent.kt
package ru.dargen.evoplus.event.chat

import net.minecraft.text.Text
import ru.dargen.evoplus.event.CancellableEvent
import ru.dargen.evoplus.util.kotlin.KotlinOpens

@KotlinOpens
data class ChatReceiveEvent(val message: Text, val isOverlay: Boolean) : CancellableEvent()